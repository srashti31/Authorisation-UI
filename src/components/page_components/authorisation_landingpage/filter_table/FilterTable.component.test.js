import React from "react";
import { configure, mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import FilterTable from "./FilterTable.component";

configure({ adapter: new Adapter() });

describe("FilterTable Component", () => {
    it("should match snapshot", () => {
        const snapshot = renderer.create(<FilterTable />).toJSON();
        expect(snapshot).toMatchSnapshot();
    });
    it("should render without crashing", () => {
        mount(<FilterTable />);
    });
    it("should able to filter", () => {
        const mockFilterCallback = (value) => {
            expect(value).toEqual("ALL");
        };
        const wrapper = shallow(
            <FilterTable filterCallback={mockFilterCallback} filterValue={"ALL"} />
        );
        let filter = wrapper.find("select.filter-option");
        filter.simulate("change", { target: { value: "ALL" } });
    });
});