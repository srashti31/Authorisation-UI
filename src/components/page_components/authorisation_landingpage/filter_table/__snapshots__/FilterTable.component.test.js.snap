// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTable Component should match snapshot 1`] = `
<div
  className="filter-content"
>
  <div
    className="filter-text"
  >
    Filter:
  </div>
  <div
    className="filter-box"
  >
    <select
      className="filter-option"
      onChange={[Function]}
    >
      <option
        value="ALL"
      >
        All
      </option>
      <option
        value="Active"
      >
        Active
      </option>
      <option
        value="Inactive"
      >
        Inactive
      </option>
    </select>
  </div>
</div>
`;
